wire-ios/Wire-iOS/Sources/UserInterface/BlockerViewController.swift

```diff
@@ -100,6 +106,16 @@ final class BlockerViewController: LaunchImageViewController {
 
         databaseFailureAlert.addAction(settingsAction)
 
+        let retryAction = UIAlertAction(
+            title: L10n.Localizable.Databaseloadingfailure.Alert.retry,
+            style: .default,
+            handler: { [weak self] _ in
+                self?.sessionManager?.recoverDatabase()
```

Comment:
what does this do? show again the popup?

Reply:
It is there just for allowing to access AppStateCalculator indirectly. It is triggered by sessionManager via delegate. So the function just runs the delegate which would then generate a new state as a result.
I've done it that way to not store & pass around completion handlers via enums as then it would be difficult to compare states in tests. It is also the cleaner option.

Reply:
ok